lms/djangoapps/courseware/toggles.py

```diff
@@ -148,6 +148,31 @@ def courseware_mfe_is_visible(
     return courseware_mfe_is_active(course_key)
 
 
+def courseware_mfe_is_advertised(
+        course_key: CourseKey,
+        is_global_staff=False,
+        is_course_staff=False,
+) -> bool:
+    """
+    Should we invite the user to view a course run's content in the Learning MFE?
+
+    This check is slightly different than `courseware_mfe_is_visible`, in that
+    we always *permit* global staff to view MFE content (assuming it's deployed),
+    but we do not shove the New Experience in their face if the preview isn't
+    enabled.
+    """
+    # DENY: Old Mongo courses don't work in the MFE.
+    if course_key.deprecated:
```

Comment:
Nit: This code is *very* close to being identical to `courseware_mfe_is_visible()`. Maybe that method could be changed to accomodate this case as well? With another passed-in boolean? Or maybe have two differently-named methods that call a common method, supplying a boolean differentiating the behavior?

Reply:
@doctoryes Yeah, I noticed the duplication, and thought that a fully-factored version might be less readable than the flat-but-non-DRY version of the two functions.

Still, I took a stab a factoring out the common logic, and I think it came out alright. I pushed it in a second commit. Let me know which you prefer, or if you'd rather I take a different approach.

Reply:
Nit: Here's more what I was thinking - but the decision will be yours at this point. I'm going to stop obsessing! 😸 :
https://github.com/edx/edx-platform/compare/kdmccormick/view-in-new-iff-preview-enabled...juliasq/dry_temp?expand=1

Reply:
Hmm, I see. Looking at all three versions, I think I'm going to stick with the original (un-factored) one in the name of simplicity.